### PR TITLE
Update CacheTagHelper.md

### PR DESCRIPTION
**Pull request by pkellner**
Fixed the vary-by-user explanation to be more clear and removed the "Needs Review" comment

### CODE REVIEW

File: aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md [hunk 1]
```diff
@@ -218,7 +218,7 @@ Example:
 </Cache>
 ```
 
-Using the attribute `vary-by-user` maintains the contents in cache through a log-in and log-out cycle.  When using `vary-by-cookie` which references the `.AspNetCore.Identity.Application` as shown above, a log-in and log-out action invalidates the cache for the same authenticated user (because a new cookie value is generated). If no user is authenticated, the state is considered a valid. [REVIEW, don't understand this] This means that no logged-in user is one cache state, and the contents will be maintained for that condition as well.
+Using this attribute maintains the contents in cache through a log-in and log-out cycle.  When using `vary-by-user`, a log-in and log-out action invalidates the cache for the same authenticated user.  This is because a new unique cookie value is generated upon login. Cache is maintained also for the anonymous state where no cookie is present or has expired. This means if no user is logged in, cache will be maintained for that condition as well.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md]: Using this attribute maintains the contents in cache through a log-in and log-out cycle.  When using `vary-by-user="true"`, a log-in and log-out action invalidates the cache for the authenticated user.  The cache is invalidated because a new unique cookie value is generated on login. Cache is maintained for the anonymous state when no cookie is present or has expired. This means if no user is logged in, the cache will be maintained.

Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md]: Link 198 has and extra ```

Comment (pkellner) [inline aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md]: @Rick-Anderson did you update this or are you wanting me to make the changes and do another pull request? It looks like a comment only but not sure.

**Comment by Rick-Anderson** [comment]
@pkellner  yes, it's only a comment.  Use it to update your PR.  You might want to tweak it a bit. Make sure it's still accurate

File: aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md (hunk 1 above)
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md]: @pkellner  if you're happy with my change I can commit it to this PR and merge. Just wanted to give you a chance to accept/edit my changes.

Comment (pkellner) [inline aspnetcore/mvc/views/tag-helpers/built-in/CacheTagHelper.md]: @Rick-Anderson yes please. I like your fixes.
